sdk/core/azure-core/src/md5.cpp

```diff
@@ -97,14 +97,14 @@ namespace Azure { namespace Core {
     }
   }
 
-  Md5::~Md5()
+  Md5Hash::~Md5Hash()
   {
     Details::Md5HashContext* md5Context = static_cast<Details::Md5HashContext*>(m_md5Context);
     BCryptDestroyHash(md5Context->hashHandle);
     delete md5Context;
```

Comment:
Are we sure that we will always have md5Context as a valid ptr to be casted?
Can it happen that the ptr was never assigned to memory addrress and the we are trying to use and delete a null ptr here?

If so, maybe add `if(md5Context)` before this

Reply:
> Are we sure that we will always have md5Context as a valid ptr to be casted?

Yes, it's initialized as part of the object ctor.